pkg/controller/atlasdeployment/atlasdeployment_controller.go

```diff
@@ -111,181 +107,123 @@ func (r *AtlasDeploymentReconciler) Reconcile(context context.Context, req ctrl.
 	}
 
 	project := &akov2.AtlasProject{}
-	if result := r.readProjectResource(context, deployment, project); !result.IsOk() {
-		workflowCtx.SetConditionFromResult(api.DeploymentReadyType, result)
+	if result = r.readProjectResource(context, atlasDeployment, project); !result.IsOk() {
+		ctx.SetConditionFromResult(api.DeploymentReadyType, result)
 		return result.ReconcileResult(), nil
 	}
 
-	if err := validate.DeploymentSpec(&deployment.Spec, r.AtlasProvider.IsCloudGov(), project.Spec.RegionUsageRestrictions); err != nil {
-		result := workflow.Terminate(workflow.Internal, err.Error())
-		workflowCtx.SetConditionFromResult(api.ValidationSucceeded, result)
+	if err := validate.AtlasDeployment(atlasDeployment, r.AtlasProvider.IsCloudGov(), project.Spec.RegionUsageRestrictions); err != nil {
+		result = workflow.Terminate(workflow.Internal, err.Error())
+		ctx.SetConditionFromResult(api.ValidationSucceeded, result)
 		return result.ReconcileResult(), nil
 	}
-	workflowCtx.SetConditionTrue(api.ValidationSucceeded)
+	ctx.SetConditionTrue(api.ValidationSucceeded)
 
-	if !r.AtlasProvider.IsResourceSupported(deployment) {
-		result := workflow.Terminate(workflow.AtlasGovUnsupported, "the AtlasDeployment is not supported by Atlas for government").
+	if !r.AtlasProvider.IsResourceSupported(atlasDeployment) {
+		result = workflow.Terminate(workflow.AtlasGovUnsupported, "the AtlasDeployment is not supported by Atlas for government").
 			WithoutRetry()
-		workflowCtx.SetConditionFromResult(api.DeploymentReadyType, result)
+		ctx.SetConditionFromResult(api.DeploymentReadyType, result)
 		return result.ReconcileResult(), nil
 	}
 
-	atlasClient, orgID, err := r.AtlasProvider.Client(workflowCtx.Context, project.ConnectionSecretObjectKey(), log)
+	atlasClient, orgID, err := r.AtlasProvider.Client(ctx.Context, project.ConnectionSecretObjectKey(), log)
 	if err != nil {
-		result := workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err.Error())
-		workflowCtx.SetConditionFromResult(api.DeploymentReadyType, result)
+		result = workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err.Error())
+		ctx.SetConditionFromResult(api.DeploymentReadyType, result)
 		return result.ReconcileResult(), nil
 	}
-	workflowCtx.OrgID = orgID
-	workflowCtx.Client = atlasClient
+	ctx.OrgID = orgID
+	ctx.Client = atlasClient
 
-	atlasSdkClient, _, err := r.AtlasProvider.SdkClient(workflowCtx.Context, project.ConnectionSecretObjectKey(), log)
+	atlasSdkClient, _, err := r.AtlasProvider.SdkClient(ctx.Context, project.ConnectionSecretObjectKey(), log)
 	if err != nil {
 		result := workflow.Terminate(workflow.AtlasAPIAccessNotConfigured, err.Error())
-		workflowCtx.SetConditionFromResult(api.DeploymentReadyType, result)
-		return result.ReconcileResult(), nil
-	}
-	workflowCtx.SdkClient = atlasSdkClient
-
-	// Allow users to specify M0/M2/M5 deployments without providing TENANT for Normal and Serverless deployments
-	r.verifyNonTenantCase(deployment)
```

Comment:
Are we not going to support that anymore?

Reply:
After a short discussion with @helderjs we decided to leave this as is and create a follow up PR to resolve this issue